x/expiration/keeper/keeper.go

```diff
@@ -129,81 +129,102 @@ func (k Keeper) GetExpiration(ctx sdk.Context, moduleAssetID string) (*types.Exp
 	}
 
-	// emit event indicating that a deposit was collected
-	return k.emitEvent(ctx, types.NewEventExpirationDeposit(expiration.ModuleAssetId, expiration.Owner, expiration.Deposit))
+	return nil
 }
 
+// ExtendExpiration extends an expiration to time in the future
 func (k Keeper) ExtendExpiration(ctx sdk.Context, expiration types.Expiration) error {
-	// get key prefix
-	key, err := types.GetModuleAssetKeyPrefix(expiration.ModuleAssetId)
+	// get key
+	key, err := types.GetModuleAssetKey(expiration.ModuleAssetId)
 	if err != nil {
-		return err
+		return types.ErrExtendExpiration.Wrapf("%s [%s]: %s",
+			types.ErrInvalidKey.Error(), expiration.ModuleAssetId, err.Error())
 	}
 
 	// lookup old expiration
 	oldExpiration, err := k.GetExpiration(ctx, expiration.ModuleAssetId)
 	if err != nil {
-		return err
+		return types.ErrExtendExpiration.Wrap(err.Error())
+	}
+	if oldExpiration == nil {
+		return types.ErrExtendExpiration.Wrapf("%s: %s", types.ErrNotFound.Error(), expiration.ModuleAssetId)
 	}
 
-	// Make sure that the new block height is higher than the old block height
+	// Validate expiration time is after old expiration time
 	if oldExpiration.Time.After(expiration.Time) {
-		k.Logger(ctx).Error("new expiration time must be after old expiration time", "err", err, "expiration", expiration.String(), "oldExpiration", oldExpiration.String())
-		return types.ErrExtendExpiration
+		return types.ErrExtendExpiration.Wrapf("new expiration time [%s] must be after old expiration time [%s]",
+			expiration.Time.UTC(), oldExpiration.Time.UTC())
 	}
 	// Validate owners are the same
 	if expiration.Owner != oldExpiration.Owner {
-		k.Logger(ctx).Error("new owner and old owner do not match", "err", err, "expiration", expiration.String(), "oldExpiration", oldExpiration.String())
-		return types.ErrNewOwnerNoMatch
+		return types.ErrNewOwnerNoMatch.Wrapf("new owner [%s] and old owner [%s] do not match",
+			expiration.Owner, oldExpiration.Owner)
 	}
 
 	// Marshal expiration record and store
 	store := ctx.KVStore(k.storeKey)
 	b, err := k.cdc.Marshal(&expiration)
 	if err != nil {
-		return err
+		return types.ErrExtendExpiration.Wrapf("%s: %s", types.ErrUnmarshal.Error(), err.Error())
 	}
 	store.Set(key, b)
 
 	// emit Extend event
 	extendEvent := types.NewEventExpirationExtend(expiration.ModuleAssetId)
-	return k.emitEvent(ctx, extendEvent)
+	err = k.emitEvent(ctx, extendEvent)
+	if err != nil {
+		return types.ErrExtendExpiration.Wrapf(
+			"failed to emit EventExpirationExtend for [%s]: %s", expiration.ModuleAssetId, err.Error())
+	}
+
+	return nil
 }
 
-// deleteExpiration removes an expiration record from the kvstore.
-func (k Keeper) deleteExpiration(ctx sdk.Context, moduleAssetID string) error {
-	key, err := types.GetModuleAssetKeyPrefix(moduleAssetID)
+// removeExpiration removes an expiration record from the kvstore.
+func (k Keeper) removeExpiration(ctx sdk.Context, moduleAssetID string) error {
```

Comment:
Can you give more context as to how it would be better?  I only left this out for easier legibility in `InvokeExpiration`. If we make it public, we create a way for modules to bypass `InvokeExpiration`. This would result in `orphan` deposits, would it not? Also, shouldn't we handle it through the `MsgServerRouter`? It would  better isolate our modules.

Reply:
Currently the flow is as follows. InvokeExpiration is triggered through a `tx` which in turn triggers the module message through the `MsgServerRouter` to delete the module asset. Once the module asset is deleted the expiration record is deleted. I'm moving removeExpiration functionality inside InvokeExpiration to eliminate confusion on whether it should be public.